lib/1.2/dml12-compatibility.dml

```diff
@@ -12,34 +12,97 @@ template bool_attr is attribute {
     param val = this;
     // Since we are by definition backed by 1.2 builtins, get throws
     shared method get() -> (attr_value_t);
+    method get() -> (attr_value_t) default {
+        return SIM_make_attr_boolean(get_val());
+    }
     shared method set(attr_value_t val) throws;
+    method set(attr_value_t val) throws default {
+        set_val(SIM_attr_boolean(val));
+    }
+    shared method get_val() -> (bool);
+    method get_val() -> (bool) default {
+        return this;
+    }
+    shared method set_val(bool val) throws;
+    method set_val(bool val) throws default {
+        this = val;
+    }
 }
 
 template uint64_attr is attribute {
     param allocate_type = "uint64";
     param val = this;
     shared method get() -> (attr_value_t);
+    method get() -> (attr_value_t) default {
+        return SIM_make_attr_uint64(get_val());
+    }
     shared method set(attr_value_t val) throws;
+    method set(attr_value_t val) throws default {
+        if (!SIM_attr_is_uint64(val)) {
+            SIM_c_attribute_error("negative integer value: %lld",
+                                  cast(SIM_attr_integer(val), int64));
+            throw;
+        }
```

Comment:
These additional checks in `uint64_attr` and `int64_attr` are ripped from 1.4 dml-builtins -- I'm guessing they were omitted from dml12-compatibility for a reason. If so, these checks should be stripped away, again; but if there was no reason, I see no reason not to introduce them. @mandolaerik ?

Reply:
There was no reason for the discrepancy, I think I just created these templates with minimal glue. The error will potentially hit errors either when they move to 1.4-style templates or when they move the main file to dml 1.4. Both are OK, but the sooner the better I suppose.